File: algorithms/strings/two-strings/two-strings-javascript/js/testeTwoStrings.js


import {twoStrings} from '../js/twoStrings.js';
import assert from 'assert';

describe('twoStrings', function() {

    it('As strings compartilharem uma substring comum YES', function() {

        let s1 = 'hello';
        let s2 = 'world';

        let esperado = 'YES';

        let resultado = twoStrings(s1, s2);

        assert.strictEqual(resultado, esperado);
    });
});

describe('twoStrings', function() {

    it('As strings compartilharem uma substring comum NO', function() {

        let s1 = 'hi';
        let s2 = 'world';

        let esperado = 'NO';

        let resultado = twoStrings(s1, s2);

        assert.strictEqual(resultado, esperado);
    });
});

describe('twoStrings', function() {

    it('As strings compartilharem uma substring comum NO', function() {

        let s1 = 'wouldyoulikefries';
        let s2 = 'abcabcabcabcabcabc';

        let esperado = 'NO';

        let resultado = twoStrings(s1, s2);

        assert.strictEqual(resultado, esperado);
    });
});

File: algorithms/strings/two-strings/two-strings-java/two-strings-java/src/TesteTwoStrings.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class TesteTwoStrings {
    
    @Test
    public void testeCaso1() {

        String s1 = "hi";
        String s2 = "world";

        String esperado = "NO";

        String resultado = Resultado.twoStrings(s1, s2);

        assertEquals(resultado, esperado);
    }

    @Test
    public void testeCaso2() {

        String s1 = "wouldyoulikefries";
        String s2 = "abcabcabcabcabcabc";

        String esperado = "NO";

        String resultado = Resultado.twoStrings(s1, s2);

        assertEquals(resultado, esperado);
    }

    @Test
    public void testeCaso3() {

        String s1 = "hello";
        String s2 = "world";

        String esperado = "YES";

        String resultado = Resultado.twoStrings(s1, s2);

        assertEquals(esperado, resultado);
    }
}
